lib/modules/versioning/types.ts

```diff
@@ -125,6 +125,11 @@ export interface VersioningApi {
    */
   subset?(subRange: string, superRange: string): boolean | undefined;
 
+  /**
+   * Checks whether subRange intersects superRange.
+   */
+  intersects?(subRange: string, superRange: string): boolean;
```

Comment:
Do we really need this on public versioning api? This is a very specific use case, so i would suggest to not do this

Reply:
I did this because it's the same approach used for the `subset()` function. I though about directly calling `semver.intersects()`, but then I would need to export the `composer2npm()` function, which doesn't seem appropriate.

Reply:
i've now seen the discussion

https://github.com/renovatebot/renovate/pull/34218/files#r1955589318